src/vs/workbench/contrib/audioCues/browser/audioCueContribution.ts

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IWorkbenchContribution } from 'vs/workbench/common/contributions';
+import { IDebugService } from 'vs/workbench/contrib/debug/common/debug';
+import { DisposableStore } from 'vs/base/common/lifecycle';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+import { Event } from 'vs/base/common/event';
+import { isCodeEditor } from 'vs/editor/browser/editorBrowser';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { raceTimeout } from 'vs/base/common/async';
+import { FileAccess } from 'vs/base/common/network';
+import { IAccessibilityService } from 'vs/platform/accessibility/common/accessibility';
+
+export class AudioCueContribution extends DisposableStore implements IWorkbenchContribution {
+	constructor(
+		@IDebugService readonly debugService: IDebugService,
+		@IEditorService readonly editorService: IEditorService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+		@IAccessibilityService private readonly accessibilityService: IAccessibilityService,
+	) {
+		super();
+
+		this.add(Event.runAndSubscribeWithStore(editorService.onDidActiveEditorChange, (_, store) => {
+			let lastLineNumber = -1;
+
+			const activeTextEditorControl = editorService.activeTextEditorControl;
+			if (isCodeEditor(activeTextEditorControl)) {
```

Comment:
Not sure if we need to account for diff editors too? I think I can debug a right hand side diff editor.